### PR TITLE
TSK-1716: Replaced immutable Collections with mutable ones in all our…

### PR DESCRIPTION
**Pull request by martisaw**
… entities.

---
 
### For the submitter:
- [ ] I updated the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) and will supply links to the specific files
- [x] I did not update the [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview)
- [x] I included a link to the [SonarCloud branch analysis](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1019969636/SonarCloud+Integration)
- [ ] I added a description of changes on the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I did not update the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana)
- [x] I put the ticket in review
- [ ] After integration of the pull request, I verified our [bluemix test environment](http://taskana.mybluemix.net/taskana) is not broken

### Verified by the reviewer:
- [x] Commit message format → TSK-XXX: Your commit message.
- [x] Submitter's update to [documentation](https://taskana.atlassian.net/wiki/spaces/TAS/overview) is sufficient
- [x] SonarCloud analysis meets our standards
- [x] Update of the [current release notes](https://taskana.atlassian.net/wiki/spaces/TAS/pages/1281392672/Current+Release+Notes+Taskana) reflects changes
- [ ] PR fulfills the ticket
- [ ] Edge cases and unwanted side effects are tested
- [x] Readability

### CODE REVIEW

**Comment by martisaw** [comment]
https://sonarcloud.io/dashboard?branch=tsk-1716&id=martisaw_taskana

**Comment by benjamineckstein** [comment]
Hey Martin, 
I know it's a trivial change, but please add a test that verifies that this collection is not immutable anymore. (This also prevents that with the next refactoring we would change it back to immutable). 

I would also like to know why this change is required for a private field. Is the actual problem that we cannot modify this map/list? Maybe we should wrap the "add" & "put" option to allow a modification but then just creating a new immutable map and this way keeping the immutability.

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java [hunk 1]
```diff
@@ -21,8 +20,8 @@
 public class TaskImpl extends TaskSummaryImpl implements Task {
 
   // All objects have to be serializable
-  private Map<String, String> customAttributes = Collections.emptyMap();
-  private Map<String, String> callbackInfo = Collections.emptyMap();
+  private Map<String, String> customAttributes = new HashMap<>();
+  private Map<String, String> callbackInfo = new HashMap<>();
```
Comment (benjamineckstein) [inline lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java:24]: Hey Martin,
I know it's a trivial change, but please add a test that verifies that this collection is not immutable anymore. (This also prevents that with the next refactoring we would change it back to immutable).

I would also like to know why this change is required for a private field. Is the actual problem that we cannot modify this map/list? Maybe we should wrap the "add" & "put" option to allow a modification but then just creating a new immutable map and this way keeping the immutability.

**Review by benjamineckstein — CHANGES_REQUESTED** [review]

File: lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java (hunk 1 above)
Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java:24]: Yes. Our users want to use `task.getCustomAttributeMap().add(XYZ)`. This is not possible with immutable maps.
There was also a long slack discussion about this.

Comment (benjamineckstein) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java:24]: Then this is cleared, but please still add some tests. 

They should be incredibly easy ... 
    `assertThatCode(() -> task.getCustomAttributeMap().add("XYZ")).doesNotThrowAnyException();`

Comment (mustaphazorgati) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java:24]: :+1:

Comment (martisaw) [reply lib/taskana-core/src/main/java/pro/taskana/task/internal/models/TaskImpl.java:24]: Hi, so I added the test. Is this enough or should I check that every mutable list we have in our entities gets such a test?

**Pull request merged**